public/controllers/register-agent/containers/register-agent/register-agent.tsx

```diff
@@ -0,0 +1,240 @@
+import { ErrorHandler } from '../../../../react-services/error-management';
+import { getMasterRemoteConfiguration } from '../../../agent/components/register-agent-service';
+import './register-agent.scss';
+import { Steps } from '../steps/steps';
+import { InputForm } from '../../../../components/common/form';
+import { getGroups } from '../../services/register-agent-services';
+import { useForm } from '../../../../components/common/form/hooks';
+import { FormConfiguration } from '../../../../components/common/form/types';
+import { useSelector } from 'react-redux';
+import { withReduxProvider } from '../../../../components/common/hocs';
+import GroupInput from '../../components/steps-three/group-input';
+import { OsCard } from '../../components/step-one/os-card/os-card';
+import {
+  ValidateServerAddress,
+  ValidateAgentName,
+} from '../../utils/validations';
+
+export const RegisterAgent = withReduxProvider(
+  ({ getWazuhVersion, hasAgents, addNewAgent, reload }) => {
+    const configuration = useSelector(
+      (state: { appConfig: { data: any } }) => state.appConfig.data,
+    );
+
+    const [wazuhVersion, setWazuhVersion] = useState('');
+    const [udpProtocol, setUdpProtocol] = useState<boolean | null>(false);
+    const [connectionSecure, setConnectionSecure] = useState<boolean | null>(
+      true,
+    );
+    const [haveUdpProtocol, setHaveUdpProtocol] = useState<boolean | null>(
+      false,
+    );
+    const [haveConnectionSecure, setHaveConnectionSecure] = useState<
+      boolean | null
+    >(false);
+    const [loading, setLoading] = useState(false);
+    const [wazuhPassword, setWazuhPassword] = useState('');
+    const [groups, setGroups] = useState([]);
+    const [needsPassword, setNeedsPassword] = useState<boolean | null>(false);
+    const [hidePasswordInput, setHidePasswordInput] = useState<boolean | null>(
+      false,
+    );
+
+    const initialFields: FormConfiguration = {
+      operatingSystemSelection: {
+        type: 'custom',
+        initialValue: '',
+        component: props => {
+          return <OsCard {...props} />;
+        },
+        options: {
+          groups,
+        },
+      },
+
+      //IP: This is a set of four numbers, for example, 192.158.1.38. Each number in the set can range from 0 to 255. Therefore, the full range of IP addresses goes from 0.0.0.0 to 255.255.255.255
+      // O ipv6: 2001:0db8:85a3:0000:0000:8a2e:0370:7334
+
+      // FQDN: Maximum of 63 characters per label.
+      // Can only contain numbers, letters and hyphens (-)
+      // Cannot begin or end with a hyphen
+      // Currently supports multilingual characters, i.e. letters not included in the English alphabet: e.g. á é í ó ú ü ñ.
+      // Minimum 3 labels
+
+      serverAddress: {
+        type: 'text',
+        initialValue: configuration['enrollment.dns'] || '',
+        validate: value => ValidateServerAddress(value),
```

Comment:
**suggestion**: you could define the `validate`

```ts
validate: ValidateServerAddress(value)
```

the `ValidateServerAddress` should start in lowercase because it is a function.